modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/inbound/InboundQueueEvent.java

```diff
@@ -235,19 +242,41 @@ public void prepareForCheckIfQueueDeletable(AndesContextInformationManager conte
     }
 
     /**
-     * Returns whether the queue can be deletable on not 
-     * @return True if deletable and wise versa
+     * Returns whether the queue can be deleted or not
+     *
+     * @return True if deletable and vice versa
      * @throws AndesException
      */
     public boolean IsQueueDeletable() throws AndesException {
         try {
-            return isQueueDeletable.get();
+            return isEventComplete.get(1000, TimeUnit.MILLISECONDS);
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
         } catch (ExecutionException e) {
-            throw new AndesException("Error occurred while checking is queue deletable", e);
+            throw new AndesException("Error occurred while checking is queue: " + queueName + " deletable", e);
+        } catch (TimeoutException e) {
+            throw new AndesException("Error occurred while checking is queue: " + queueName + " deletable", e);
         }
         return false;
     }
 
+    /**
+     * Wait until the queue is added
+     */
+    public void waitForCompletion() throws AndesException {
+        try {
+            //stay blocked until the queue addition is complete
+            isEventComplete.get(1000, TimeUnit.MILLISECONDS);
```

Comment:
When I tested in the standalone mode, it took only a few milliseconds. But in a loaded envronment this could take a longer time than that. Therefore, increased the waiting time for 5 seconds. Also made the waiting time a constant. Both are addressed by https://github.com/wso2/andes/pull/381